test/graph_rewrites/enter_in_catalog_test.cc

```diff
@@ -0,0 +1,273 @@
+  ASSERT_OK(EncapsulateClusters(&graph, 0, fdeflib_new, {}));
+  ASSERT_OK(EnterInCatalog(&graph, 0));
+
+  bool remove = false;
+  for (auto node : graph.op_nodes()) {
+    auto node_name = node->name();
+    remove = false;
+    if (node_name == "Assign") {
+      ASSERT_NOT_OK(GetNodeAttr(node->attrs(), "_ngraph_remove", &remove));
+      ASSERT_FALSE(remove);
+    } else if (node_name == "Var_Assign") {
+      ASSERT_OK(GetNodeAttr(node->attrs(), "_ngraph_remove", &remove));
+      ASSERT_TRUE(remove);
+    }
+  }
+}
+
+//          Var_A      Const     Var_B
+//          /|  \       /         | \
+//         / |   \     /          |  \
+//  Assign_A |    \   /           |  Assign_B
+//           |     Add            |
+//           |      /\            |
+//           |     /  \           |
+//           |    /    \          |
+//          Assign_1   Acos       |
+//                       \        |
+//                        \       |
+//                         \      |
+//                         Assign_2
+// Assign_A, Assign_B and Assign_1 should have the attribute
+// _ngraph_remove added and set to true
+// whereas Assign_2 should not have the attribute added since
+// it is being fed by Acos which is not supported by nGraph.
+TEST(CatalogTest, SmallGraph3) {
+  Scope root = Scope::NewRootScope();
+
+  PartialTensorShape varShape({2, 2});
+  auto var_a = ops::Variable(root.WithOpName("VarA"), varShape, DT_FLOAT);
+  auto init_value_a = ops::Const(root, {{1.f, 1.f}, {1.f, 1.f}});
+  auto var_assign_a =
+      ops::Assign(root.WithOpName("VarA_Assign"), var_a, init_value_a);
+
+  auto var_b = ops::Variable(root.WithOpName("VarB"), varShape, DT_FLOAT);
+  auto init_value_b = ops::Const(root, {{2.f, 2.f}, {2.f, 2.f}});
+  auto var_assign_b =
+      ops::Assign(root.WithOpName("VarB_Assign"), var_b, init_value_b);
+
+  auto c = ops::Const(root, {{1.f, 1.f}, {1.f, 1.f}});
+
+  auto add = ops::Add(root.WithOpName("Add"), var_a, c);
+
+  auto assign_1 = ops::Assign(root.WithOpName("Assign_1"), var_a, add);
+
+  auto acos = ops::Acos(root.WithOpName("Acos"), add);
+
+  auto assign_2 = ops::Assign(root.WithOpName("Assign_2"), var_b, acos);
+
+  std::set<string> skip_these_nodes = {};
+
+  Graph graph(OpRegistry::Global());
+  TF_CHECK_OK(root.ToGraph(&graph));
+
+  // Execute all the passes one by one
+  ASSERT_OK(CaptureVariables(&graph, skip_these_nodes));
+  ASSERT_OK(ReplaceModifiers(&graph, 0));
+  ASSERT_OK(MarkForClustering(&graph, skip_these_nodes, "CPU"));
+  ASSERT_OK(AssignClusters(&graph));
+  FunctionDefLibrary* fdeflib_new = new FunctionDefLibrary();
+  ASSERT_OK(EncapsulateClusters(&graph, 0, fdeflib_new, {}));
+  ASSERT_OK(EnterInCatalog(&graph, 0));
+
+  bool remove = false;
+  for (auto node : graph.op_nodes()) {
+    auto node_name = node->name();
+    remove = false;
+    if (node_name == "VarA_Assign") {
+      ASSERT_OK(GetNodeAttr(node->attrs(), "_ngraph_remove", &remove));
```

Comment:
We can check ExistsInCatalo for these ops too. As it is a more complex graph

Reply:
I wanted to keep it separate, but we can do it too.